tests/Dfe.PlanTech.AzureFunctions.UnitTests/Functions/QueueReceiverTests.cs

```diff
@@ -343,15 +344,21 @@ await serviceBusMessageActionsMock.Received()
         Assert.True(_questions[1].Deleted);
     }
 
-    [Fact]
-    public async Task QueueReceiverDbWriter_Should_ExitEarly_When_Event_Is_Save_And_UsePreview_Is_False()
+    [Theory]
+    [InlineData("ContentManagement.Entry.save")]
+    [InlineData("ContentManagement.Entry.auto_save")]
+    public async Task QueueReceiverDbWriter_Should_ExitEarly_When_Event_Is_Save_And_Entity_Is_Published_And_UsePreview_Is_False(string subject)
     {
+        _contentComponent.Published = true;
+        _questions.Add(_contentComponent);
+        _entityRetrieverMock
+            .When(mock => mock.GetExistingDbEntity(Arg.Any<ContentComponentDbEntity>(), default)
+                .Returns(_contentComponent));
```

Comment:
This could be improved by checking what arguments the call received, then only returning `_contentComponent` if the IDs match.

Reply:
good shout, done